### PR TITLE
(Core\CMD) Npc Support for Freeze and Unfreeze

### PR DESCRIPTION
**Pull request by acidmanifesto**
**Changes proposed:**

- Remove Magic Numbers dealing with 9454 (Freeze Aura)
- Added Support for GM Cmd .freeze and .unfreeze to target and work on  NPCS

**Issues addressed:**

Closes #  No Known Issues

**Tests performed:**

- [x] Builds.
- [x] Tested in Game.

**Steps to Test:**

1. Target moving npcs and type .freeze to freeze the npcss.
2. Target moving npcs and type .unfreeze to unfreeze the npcs with the frozen aura.

### CODE REVIEW

**Comment by MaxtorCoder** [comment]
Oof, plz change indent settings and reformat the file 😄

**Comment by acidmanifesto** [comment]
> Oof, plz change indent settings and reformat the file 😄

i just have github lie to me for my confidence and i select hide whitespace changes. :)
Honestly. not certain how the indents happened truthfully.

File: src/server/scripts/Commands/cs_misc.cpp [hunk 1]
```diff
@@ -2549,6 +2555,7 @@ class misc_commandscript : public CommandScript
     {
         std::string name;
         Player* player;
+        Unit* target = handler->getSelectedUnit();
```
Comment (Goatform) [inline src/server/scripts/Commands/cs_misc.cpp:2558]: null check?

File: src/server/scripts/Commands/cs_misc.cpp [hunk 2]
```diff
@@ -2449,6 +2454,7 @@ class misc_commandscript : public CommandScript
 
     static bool HandleFreezeCommand(ChatHandler* handler, char const* args)
     {
+        Unit* target = handler->getSelectedUnit();
```
Comment (Goatform) [inline src/server/scripts/Commands/cs_misc.cpp:2457]: check if null check is required

File: src/server/scripts/Commands/cs_misc.cpp (hunk 1 above)
Comment (acidmanifesto) [reply src/server/scripts/Commands/cs_misc.cpp:2563]: got a check in line 2594, hopefully it is what you ment. If not then im a bit confused in how you would want the null check to be done exactly in this case.

File: src/server/scripts/Commands/cs_misc.cpp (hunk 2 above)
Comment (acidmanifesto) [reply src/server/scripts/Commands/cs_misc.cpp:2457]: got a check in line 2534, hopefully it is what you ment. If not then im a bit confused in how you would want the null check to be done exactly in this case.

**Comment by jackpoz** [comment]
The if blocks are not chained together correctly.
there is `if (has aura) return; if (does not have aura) return; else`, meaning the `else` will never happen. the old code also had the `else` linked to `if (player)`. Please check this code again and make sure the cases handled before are still handled correctly.

(I think comment in the code as github is having issues)

**Comment by jackpoz** [comment]
> The if blocks are not chained together correctly. there is `if (has aura) return; if (does not have aura) return; else`, meaning the `else` will never happen. the old code also had the `else` linked to `if (player)`. Please check this code again and make sure the cases handled before are still handled correctly.
> 
> (I think comment in the code as github is having issues)

@acidmanifesto this PR has some pending review for you to deal with 😃

**Comment by acidmanifesto** [comment]
Lol. So it is. I been caught up on the anticheat wip i pretty much forgot about the others. When i get some free time outside of the anticheat obsession i will try to get back to this.

**Comment by acidmanifesto** [comment]
I am going to close this for now and reopen it another way at a later time.

**Pull request closed without merging**